src/NextGen/Gateways/Stripe/NextGenStripeGateway/NextGenStripeRepository.php

```diff
@@ -34,40 +36,52 @@ protected function generateStripePaymentIntent($accountId, Money $amount): Payme
      * Get or create Stripe Customer from Donation
      *
      * @unreleased
-     * @throws ApiErrorException
      * @throws Exception
+     * @throws ApiErrorException
      */
     public function getOrCreateStripeCustomerFromDonation(
         string $connectAccountId,
         Donation $donation
-    ): Customer {
+    ): Customer
+    {
         $donorCustomerId = give_stripe_get_customer_id($donation->email) ?? '';
 
-        // make sure customerId still exists in  connect account
-        if ($donorCustomerId) {
-            $customer = Customer::retrieve($donorCustomerId, ['stripe_account' => $connectAccountId]);
+        // create a new customer if the donor does not have a stripe id
+        if (!$donorCustomerId) {
+            $customer = $this->createCustomer($donation, $connectAccountId);
+        } else {
+            try {
+                // if the donor has a stripe ID, try retrieving the customer from Stripe.
+                $customer = $this->retrieveCustomer($donorCustomerId, $connectAccountId);
+
+                // if the customer is deleted, create a new customer
+                if ($customer->isDeleted()) {
+                    $customer = $this->createCustomer($donation, $connectAccountId);
+                }
+            } catch (InvalidRequestException $e) {
```

Comment:
Is there a more specific Exception that we could catch? Nesting an `if` inside of a `catch` can suppress instances of the Exception if not handled by and `if` condition.

Reply:
Resolved bb4817dc